Sources/DiscordKit/APIUtils.swift

```diff
@@ -7,6 +7,18 @@
 
 import Foundation
 
+let iso8601 = { () -> ISO8601DateFormatter in
+    let fmt = ISO8601DateFormatter()
+    fmt.formatOptions = [.withInternetDateTime]
+    return fmt
+}()
+
+let iso8601WithFractionalSeconds = { () -> ISO8601DateFormatter in
+    let fmt = ISO8601DateFormatter()
+    fmt.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
+    return fmt
+}()
```

Comment:
annoyingly if you specify with fractional seconds, you must use fractional seconds, and the discord api is inconsistent with this.

Reply:
Yeah thats the annoying part. Another problematic thing is the nonce value in `Message`. It can either be an int or a string according to the dev docs, so I have no idea how to add that property.

Reply:
Overall this PR seems quite well done, would be great if you could resolve the issues with #6